components/director/internal/domain/document/service_test.go

```diff
@@ -309,3 +358,115 @@ func TestService_Delete(t *testing.T) {
 		})
 	}
 }
+
+func TestService_GetFetchRequest(t *testing.T) {
+	// given
+	tnt := "tenant"
+	ctx := context.TODO()
+	ctx = tenant.SaveToContext(ctx, tnt)
+
+	testErr := errors.New("Test error")
+
+	refID := "doc-id"
+	frURL := "foo.bar"
+	timestamp := time.Now()
+
+	fetchRequestModel := fixModelFetchRequest("foo", frURL, timestamp)
+
+	testCases := []struct {
+		Name                 string
+		RepositoryFn         func() *automock.DocumentRepository
+		FetchRequestRepoFn   func() *automock.FetchRequestRepository
+		InputDocumentID      string
+		ExpectedFetchRequest *model.FetchRequest
+		ExpectedErrMessage   string
+	}{
+		{
+			Name: "Success",
+			RepositoryFn: func() *automock.DocumentRepository {
+				repo := &automock.DocumentRepository{}
+				repo.On("Exists", ctx, tnt, refID).Return(true, nil).Once()
+				return repo
+			},
+			FetchRequestRepoFn: func() *automock.FetchRequestRepository {
+				repo := &automock.FetchRequestRepository{}
+				repo.On("GetByReferenceObjectID", ctx, tnt, model.DocumentFetchRequestReference, refID).Return(fetchRequestModel, nil).Once()
+				return repo
+			},
+			InputDocumentID:      refID,
+			ExpectedFetchRequest: fetchRequestModel,
+			ExpectedErrMessage:   "",
+		},
+		{
+			Name: "Success - Not Found",
+			RepositoryFn: func() *automock.DocumentRepository {
+				repo := &automock.DocumentRepository{}
+				repo.On("Exists", ctx, tnt, refID).Return(true, nil).Once()
+				return repo
+			},
+			FetchRequestRepoFn: func() *automock.FetchRequestRepository {
+				repo := &automock.FetchRequestRepository{}
+				repo.On("GetByReferenceObjectID", ctx, tnt, model.DocumentFetchRequestReference, refID).Return(nil, repopkg.NewNotFoundError()).Once()
+				return repo
+			},
+			InputDocumentID:      refID,
+			ExpectedFetchRequest: nil,
+			ExpectedErrMessage:   "",
+		},
+		{
+			Name: "Error - Get FetchRequest",
+			RepositoryFn: func() *automock.DocumentRepository {
+				repo := &automock.DocumentRepository{}
+				repo.On("Exists", ctx, tnt, refID).Return(true, nil).Once()
+
+				return repo
+			},
+			FetchRequestRepoFn: func() *automock.FetchRequestRepository {
+				repo := &automock.FetchRequestRepository{}
+				repo.On("GetByReferenceObjectID", ctx, tnt, model.DocumentFetchRequestReference, refID).Return(nil, testErr).Once()
+				return repo
+			},
+			InputDocumentID:      refID,
+			ExpectedFetchRequest: nil,
+			ExpectedErrMessage:   testErr.Error(),
+		},
+		{
+			Name: "Error - Document doesn't exist",
```

Comment:
When document not exist, repo should return:
`false, nil`
or `false, error`?
For me the `false, nil` is better, because repo return `true, nil` if found.

Reply:
`false, nil` of course